downloads/swan-lake-release-notes/swan-lake-alpha2/RELEASE_NOTE.md

```diff
@@ -2,41 +2,554 @@
+
+- The `mail.smtp.ssl.checkserveridentity` custom property was passed as an entry in the `properties` to enable/disable the server certificate’s hostname verification. As the `properties` field is removed from the API, from this release onwards, the `verifyHostName` boolean field is introduced to the `secureSocket` record for all the configurations related to the Email module. 
+
+This `email:SecureSocket` record change would appear as follows.
+
+```ballerina
+public type SecureSocket record {|
+    // ... other fields
+    boolean verifyHostname = true;
+|};
+```
+
+##### TCP Module Improvements
+
+Introduced SSL/TLS-based communication to the TCP module using the `secureSocket` record configuration in both the client and the listener.
+
+###### TCP Client with `secureSocket`
+
+A sample code is as follows.
+
+```ballerina
+import ballerina/tcp;
+
+configurable string keyPath = ?;
+configurable string certPath = ?;
+
+public function main() returns error? {
+    tcp:Client socketClient = check new ("localhost", 9002, secureSocket = {
+        certificate: {path: certPath},
+        protocol: {
+            name: "TLS",
+            versions: ["TLSv1.2", "TLSv1.1"]
+        },
+        ciphers: ["TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA"]
+    });
+
+    string msg = "Hello Ballerina Echo from secure client";
+    byte[] msgByteArray = msg.toBytes();
+    check socketClient->writeBytes(msgByteArray);
+
+    readonly & byte[] receivedData = check socketClient->readBytes();
+    test:assertEquals('string:fromBytes(receivedData), msg, "Found unexpected output");
+
+    check socketClient->close();
+}
+```
+
+###### TCP Listener with `secureSocket`
+
+A sample code is as follows.
+
+```ballerina
+configurable string keyPath = ?;
+configurable string certPath = ?;
+
+service on new tcp:Listener(9002, secureSocket = {
+    certificate: {path: certPath},
+    privateKey: {path: keyPath},
+    protocol: {
+        name: "TLS",
+        versions: ["TLSv1.2", "TLSv1.1"]
+    },
+    ciphers: ["TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA"]
+}) {
+
+    isolated remote function onConnect(tcp:Caller caller) returns tcp:ConnectionService {
+        io:println("Client connected to secureEchoServer: ", caller.remotePort);
+        return new EchoService(caller);
+    }
+}
+
+service class EchoService {
+    tcp:Caller caller;
+
+    public isolated function init(Caller c) {
+        self.caller = c;
+    }
+
+    remote function onBytes(readonly & byte[] data) returns (readonly & byte[])|tcp:Error? {
+        io:println("Echo: ", 'string:fromBytes(data));
+        return data;
+    }
+}
+```
+
+##### UDP Module Improvements
+
+- The `sendDatagram` function will now send multiple datagrams if the size of the `byte[]` value provided as the `data` field of the datagram exceeds 8KB.
```

Comment:
This is a method in a client, right? @MohamedSabthar 
```suggestion
- The `sendDatagram` method will now send multiple datagrams if the size of the `byte[]` value provided as the `data` field of the datagram exceeds 8KB.
```

Reply:
+1